### PR TITLE
mainline uboot

### PR DESCRIPTION
**Pull request by rroohhh**
use mainline u-boot instead of the xilinx fork

Todo:
- [x] test on micro
- [x] test on beta
- [x] make qemu working again
   caveat: qemu boots the kernel directly and does not boot `u-boot` for now (until we can figure out how to make `u-boot` work again)

Maybe we also want to use falcon mode, but probably not (would boot a bit faster, but we loose all the nice u-boot features like tftp boot.

This does some minor build system rework, to
- add patch support for u-boot
- add device specific configuration (we might want to integrate some existing stuff into that)
- use `.install_stamp` for `boot.fs` aswell, as that should work better

### CODE REVIEW

**Comment by anuejn** [comment]
seems good to me at the first glance :). However, did not test it yet

**Comment by rroohhh** [comment]
So one thing I am not sure about are the u-boot `defconfig` files.

Adding them as patches makes it easy to deploy them over the already existing patches infrastructure and atleast for the `beta` (`microzed`) we get "automatic" updates to our `defconfigs` from upstream (because we patch them and don't override). 

However editing the `defconfig` files per hand is a lot easier when they are not in patch form.

**Comment by rroohhh** [comment]
booting the kernel works using
```bash
build/qemu-xilinx-v2018.3.git/aarch64-softmmu/qemu-system-aarch64 -M arm-generic-fdt-7series -serial /dev/null -serial mon:stdio -display none  -nographic -dtb build/boot.fs/devicetree.dtb -drive if=sd,format=raw,index=0,file=build/axiom-micro.img -kernel build/boot.fs/zImage -append 'console=ttyPS0,115200n8 root=PARTUUID=f37043ff-02 rw rootfstype=ext4 rootwait systemd.log_level=warning loglevel=7 systemd.log_target=console kernel.sysrq=1 init=/usr/lib/systemd/systemd sdhci.debug_quirks=64 kernel.sysrq=1'
```

However I can't get booting u-boot to works. Maybe we don't need qemu emulation of `u-boot`?

**Comment by rroohhh** [comment]
So after testing this on the `beta` I think this is good to go.

The `u-boot` qemu problem is not really that bad, as nothing is currently using that capability.

File: .cirrus.yml [hunk 1]
```diff
@@ -13,8 +13,10 @@ docker_builder:
       git fetch origin pull/$CIRRUS_PR/head:pull/$CIRRUS_PR
     fi;
     git reset --hard $CIRRUS_CHANGE_IN_REPO
-  make_script: ./makefiles/docker-make.sh DEVICE=$DEVICE
-  test_script: ./makefiles/docker-make.sh DEVICE=$DEVICE test
+  make_script: >
+      ./makefiles/docker-make.sh DEVICE=$DEVICE
+  test_script: >
+      ./makefiles/docker-make.sh DEVICE=$DEVICE test
```
Comment (anuejn) [inline .cirrus.yml]: y tho?

Comment (rroohhh) [reply .cirrus.yml]: artifact from previous commits, will remove

**Comment by rroohhh** [comment]
will merge after the ci runs through, but the work here is completely done, see #105 and #106.

**Pull request merged**